hivemind/moe/client/expert.py

```diff
@@ -1,43 +1,72 @@
-from typing import Any, Dict, Optional, Tuple
+from __future__ import annotations
+
+import os
+from concurrent.futures import Future
+from dataclasses import dataclass
+from queue import Queue
+from threading import Thread
+from typing import Any, Awaitable, Dict, Iterable, List, Optional, Sequence, Tuple, Union
 
 import torch
 import torch.nn as nn
 from torch.autograd.function import once_differentiable
 
+import hivemind
```

Comment:
Oops, you're right, my suggestion is actually making the circular import issues worse. Let's keep it as is then :)

As for `dht_handler.py`, I think `hivemind/moe/server` is a good place for it since it's an implementation detail of moe.server (not used by anything else in hivemind).